pkg/querier/tripperware/instantquery/instant_query.go

```diff
@@ -154,7 +161,7 @@ func (c instantQueryCodec) EncodeRequest(ctx context.Context, r tripperware.Requ
 		}
 	}
 
-	tripperware.SetRequestHeaders(h, c.defaultCodecType, c.compression)
+	tripperware.SetRequestHeaders(h, c.defaultCodecType, c.compression, c.rulerQueryResponseFormat)
```

Comment:
This will always ask for cortex internal response format in instant query even though the requests are not initiated from Ruler?

Reply:
oh.. I see.
We can utilize the `User-Agent` header value at `DecodeRequest` and `EncodeRequest`.